Source/Dafny/Resolver.cs

```diff
@@ -7044,6 +7068,16 @@ enum TailRecursionStatus
               tRhs.InitCall.Method.Name);
           }
           return TailRecursionStatus.NotTailRecursive;
+        } else if (s.Rhs is ExprRhs eRhs && eRhs.Expr.Resolved is FunctionCallExpr fce && fce.Function.ByMethodDecl == enclosingMethod) {
+          var status = ConfirmTailCall(s.Tok, enclosingMethod, fce.TypeApplication_JustFunction, new List<Expression>() { s.Lhs }, reportErrors);
```

Comment:
Can you write `new() { s.Lhs }` here instead?

Reply:
Cool, I didn't know such a thing was possible. Hmm, actually, no, it isn't -- it gives me an error when I try it.